### PR TITLE
Add engines versions to the URL and logic change

### PR DESCRIPTION
**Pull request by lukasz-rutkowski-red**
- Engines versions now are visible in the URL if they are not default.
- Engine change now is **not permanent**, if the testsuite tab is changed, engine version will be set to the default.

### CODE REVIEW

File: src/engineChange.js [hunk 1]
```diff
@@ -57,15 +57,24 @@ function loadStoredEngine() {
...
+      // if engine version is provided and it is the same in Local Storage do nothing:
+    } else if (queryVariable && window.localStorage["engine_" + engineId] === queryVariable) {
+      engine.defaultVersion = queryVariable;
     } else {
-      var version = window.localStorage["engine_" + engineId];
-      if (engine.versions[version] !== undefined) {
-        engine.defaultVersion = window.localStorage["engine_" + engineId];
+      // if engine version is provided but it does not exist in 'EngineVersions' dict, replace it in url to the default value:
+      if (engine.versions[queryVariable] === undefined) {
+        window.localStorage["engine_" + engineId] = engine.defaultVersion;
```
Comment (maciej-kolanski-red) [inline src/engineChange.js:70]: Local Storage is mechanism for storing data in browser session, so it is can be retrieved after window reload: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage

Right now the "engine_*" parameter is always either read from URL or default, so I believe we can drop usage of Local Storage in this context

File: src/engineChange.js [hunk 2]
```diff
@@ -57,15 +57,24 @@ function loadStoredEngine() {
     var engine = EngineVersions[engineId];
     var queryVariable = getQueryVariable("engine_" + engineId);
 
-    if (queryVariable && engine.versions[queryVariable] !== undefined) {
-      window.localStorage["engine_" + engineId] = queryVariable;
-      window.location.href = window.location.search.replace("engine_" + engineId + "=" + queryVariable, "");
-    } else if (window.localStorage["engine_" + engineId] === undefined) {
+    // if engine version is missing, add the default value from dict 'EngineVersions' to the url:
```
Comment (maciej-kolanski-red) [inline src/engineChange.js:60]: How about if no "engine_" URL is provided, then just use default version and do not set these URL parameters?

I believe that would be the simplest scenario from user POV - it works by default and URL parameters are used only if its required.

File: src/engineChange.js [hunk 3]
```diff
@@ -57,15 +57,24 @@ function loadStoredEngine() {
...
       window.localStorage["engine_" + engineId] = engine.defaultVersion;
+      window.location.href += "&engine_" + engineId + "=" + engine.defaultVersion;
+      // if engine version is provided and it is the same in Local Storage do nothing:
+    } else if (queryVariable && window.localStorage["engine_" + engineId] === queryVariable) {
+      engine.defaultVersion = queryVariable;
     } else {
-      var version = window.localStorage["engine_" + engineId];
-      if (engine.versions[version] !== undefined) {
-        engine.defaultVersion = window.localStorage["engine_" + engineId];
+      // if engine version is provided but it does not exist in 'EngineVersions' dict, replace it in url to the default value:
```
Comment (maciej-kolanski-red) [inline src/engineChange.js:68]: Please print a console warning/error in such case like:

Shaka version x.x.x is not supported, overriding with x.x.x

**Review by maciej-kolanski-red — CHANGES_REQUESTED** [review]

File: src/engineChange.js (hunk 1 above)
Comment (lukasz-rutkowski-red) [reply src/engineChange.js]: Using of Local Storage dropped.

File: src/engineChange.js (hunk 2 above)
Comment (lukasz-rutkowski-red) [reply src/engineChange.js]: Parameters are not visible in the URL now, when default parameters are chosen.

File: src/engineChange.js (hunk 3 above)
Comment (lukasz-rutkowski-red) [reply src/engineChange.js]: Console warning added.

**Review by maciej-kolanski-red — APPROVED** [review]

**Review by maciej-kolanski-red — APPROVED** [review]

**Pull request merged**